### ISSUE TITLE
Missing essential features

### ISSUE THREAD

**Issue by psy0rz** [Instathings/modbus2mqtt#11]
First of all: great initiative! 

I tried converting my project to a herdsman module for your project. (https://github.com/psy0rz/YTLmodbus)

I ran into several issues, since this will be the first module added by a third party i guess:

- My dongle or device seems to need partiy even. (i hardcoded it in lib/methods/start.js)
- While the parity was wrong there was zero output or feedback of what was going on. (setting DEBUG=1 environment didnt help, it seems there is no debugging code)
- A read error returns "undefined" in the error log. I changed logger in lib/modbus.js to: logger.error("While reading address "+address+": "+err);
- My device needs FC3 (readHoldingRegisters()), probably with a different lenght parameter. Currently the library only supports this.modbus.readInputRegisters()
- I also hardcoded that in, just to test, and it produced:
```
modbus2mqtt/m1 {"voltage":{"data":[23183],"buffer":{"type":"Buffer","data":[90,143]}}}
```

It seems the datafield has the correct number, but i probably need to do some post processing?

Could you fix the above issues and provide at least a framework that allows for different function codes (e.g. FC3), different readlengths and different parities? 

And perhaps a better example?

After that i'll try again.

**Comment by psy0rz** [comment]
Also: it crashes with an unhandled exception when sending invalid json to modbus2mqtt/configure/set

**Comment by psy0rz** [comment]
i'm fixing most of these, expect a PR :)

**Comment by psy0rz** [comment]
Fixed in #12
